docs/source/loading.rst

```diff
@@ -0,0 +1,304 @@
+
+.. code-block::
+
+    {"version": "0.1.0",
+     "data": [{"a": 1, "b": 2.0, "c": "foo", "d": false},
+              {"a": 4, "b": -5.5, "c": null, "d": true}]
+    }
+
+    >>> from datasets import load_dataset
+    >>> dataset = load_dataset('json', data_files='my_file.json', field='data')
+
+While these are the most common formats, you may encounter other datasets that are stored differently. Datasets will recognize this, and fallback accordingly on the Python ``json`` loading methods to handle the various formats.
+
+Text files
+^^^^^^^^^^
+
+Text files are one of the most common methods for storing a dataset. Datasets will read the text file line by line to build the dataset.
+
+    >>> from datasets import load_dataset
+    >>> dataset = load_dataset('text', data_files={'train': ['my_text_1.txt', 'my_text_2.txt'], 'test': 'my_test_file.txt'})
+
+In-memory data
+--------------
+
+Datasets will also allow you to create a Dataset directly from in-memory data structures. Currently, Datasets supports loading data directly from Python dictionaries and Pandas DataFrames.
+
+Python dictionary
+^^^^^^^^^^^^^^^^^
+
+Python dictionaries can be loaded using the :func:`datasets.Dataset.from_dict()` method:
+
+    >>> from datasets import Dataset
+    >>> dataset = Dataset.from_dict(my_dict)
+
+Pandas DataFrame
+^^^^^^^^^^^^^^^^
+
+Likewise, datasets in Pandas DataFrames can be instantiated by:
+
+    >>> from datasets import Dataset
+    >>> import pandas as pd
+    >>> df = pd.DataFrame({"a": [1, 2, 3]})
+    >>> dataset = Dataset.from_pandas(df)
+
+.. important::
+
+    An object data type in pandas.Series doesn't always carry enough information for Arrow to automatically infer a type. Avoid this by constructing an explicit schema with :class:`datasets.Features` using the ``from_dict`` or ``from_pandas`` methods. See the Troubleshooting guide below for more details on how to specify a feature.
+
+Streaming
+--------------
+
+Streaming a dataset is useful when you want to start using the dataset without having to wait to download the entire dataset. It also allows you to work with datasets that exceed the amount of disk space on your computer. The data is downloaded progressively as you iterate over the dataset. 
+
+Stream a dataset by setting ``streaming=True`` in ``load_dataset()`` as shown below:
+
+    >>> from datasets import load_dataset
+    >>> dataset = load_dataset('oscar', "unshuffled_deduplicated_en", split='train', streaming=True)
+    >>> print(next(iter(dataset)))
+    {'text': 'Mtendere Village was inspired by the vision of Chief Napoleon Dzombe, which he shared with John Blanchard during his first visit to Malawi. Chief Napoleon conveyed the desperate need for a program to intervene and care for the orphans and vulnerable children (OVC) in Malawi, and John committed to help...
```

Comment:
I think we should redirect to a page dedicated to streaming and add more explanations on how to use dataset streaming. Maybe add a link to the part about `IterableDataset` in process.rst ?

Reply:
Created a how-to page dedicated to streaming and processing methods associated with `IterableDataset`